java/flight/flight-sql-jdbc-core/src/test/java/org/apache/arrow/driver/jdbc/utils/ResultSetTestUtils.java

```diff
@@ -68,33 +57,39 @@ public static <T> void testData(
    * @param columnNames the column names to fetch in the {@code ResultSet} for comparison.
    * @param expectedResults the rows and columns representing the only values the {@code resultSet}
    *     is expected to have.
-   * @param collector the {@link ErrorCollector} to use for asserting that the {@code resultSet} has
-   *     the expected values.
    * @param <T> the type to be found in the expected results for the {@code resultSet}.
    * @throws SQLException if querying the {@code ResultSet} fails at some point unexpectedly.
    */
   @SuppressWarnings("unchecked")
   public static <T> void testData(
       final ResultSet resultSet,
       final List<String> columnNames,
-      final List<List<T>> expectedResults,
-      final ErrorCollector collector)
+      final List<List<T>> expectedResults)
       throws SQLException {
     testData(
         resultSet,
         data -> {
+          List<Throwable> errors = new ArrayList<>();
           final List<T> columns = new ArrayList<>();
           for (final String columnName : columnNames) {
             try {
               columns.add((T) resultSet.getObject(columnName));
             } catch (final SQLException e) {
-              collector.addError(e);
+              errors.add(e);
             }
           }
+          assertAll(
+              "Errors occurred while fetching data from the ResultSet",
+              errors.stream()
+                  .map(
+                      error ->
+                          () -> {
+                            throw error;
+                          }));
+
```

Comment:
ETC. Is this change appropriate?

Reply:
Why would we re-throw the error? I think that would short-circuit the test unexpectedly. 

That said, the version of this with an error collector is never actually used. I would rather remove the redundant overloads and let any exceptions here bubble up and terminate the test, instead of try to preserve the original semantics which aren't used anyways.